justfile

```
@@ -237,6 +237,10 @@ ipython:
     # The resulting output will be at `api/static` and is git ignored for convenience.
     @STATIC_ROOT="./static" just dj collectstatic --noinput
 
+# Launch a pgcli shell on the web container
+db-shell db_host="db":
+    docker-compose {{ DOCKER_FILE }} exec web pgcli -h {{ db_host }} openledger deploy
```

We should use the [`dbshell` management command](https://docs.djangoproject.com/en/4.1/ref/django-admin/#dbshell). It will automatically set the host, username and password for us (from the `DATABASES` setting) and drop us directly in the SQL prompt.

```suggestion
    just dj dbshell
```

It uses the `psql` command-line client so not 100% sure if it'll work inside the Docker container without adding a step to install it first.

---

It also needs an extra package installed on the `web` container, `postgresql-client`, or it throws an error:

```sh
CommandError: You appear not to have the 'psql' program installed or on your path.
error: Recipe `exec` failed on line 72 with exit code 1
error: Recipe `dj` failed on line 223 with exit code 1
```

I guess we can add it as well to avoid having to type the password, could be the `just pqsql` command, but [`pgcli`](https://github.com/dbcli/pgcli) has some nice extra features in comparison to the basic `psql`, like autocompletion and syntax highlighting 📝

I wonder if the rest of folks find valuable the extra size for this feature? @obulat @AetherUnbound

---

I'm definitely a big fan of `pgcli`, and I was about to mention that we'd probably need to use `psql` (with the code as it stands now) since we're not [installing it in our postgres image like we do for the catalog](https://github.com/WordPress/openverse-catalog/blob/main/docker/local_postgres/Dockerfile). We're [already installing some libraries in the docker image](https://github.com/WordPress/openverse-api/blob/431b2f9f71423f96cc823a6d9141017e065d3329/api/Dockerfile#L80) so I think adding the postgresql-client library would be alright 🙂 We could always create dev and prod images from the Docker build where the prod one does not have any dev libraries.

We don't necessarily have to install pgcli on the Postgres image like we do for the catalog but it would need to be installed somewhere if we wanted to use it. That to say, I'm +1 for setting it up to work with the API one way or another! It seems like we could set up the `dbshell` Django command to use `pgcli`, but it'd require an additional library: https://www.pgcli.com/django-pgcli.html

---

`django-pgcli` hasn't been updated in 2 years (not that it needs to, it's just 19 lines) so there is a possibility it might not work with the newest Django!

I sometimes use the `psql` client installed on the host to connect to the Docker `db` service and it works fine.

---

Added both commands! 🙂